### DISCUSSION TITLE
Best practices for AWS cross region mirroring use case with Strimzi MM2?

### DISCUSSION THREAD

**Discussion by iftachbyaf** [strimzi/strimzi-kafka-operator#9946] [Q&A] [answered]
Hey guys,

I wanted to create a discussion about general best practices for a use case of mirroring from multiple spoke regions in AWS to a single hub region.

We have a Kafka cluster and a K8S cluster in each spoke region and in the hub.

Initially I planned to create a separate deployment of Strimzi MM2 in each spoke region k8s that will each connect to the regional Kafka and to the hub Kafka. This is possible but due to the limitation of the MM2 Connect needing to connect to the target Kafka I'm assuming it would be ideal to have the MM2 cluster in the same region as the target? To minimize latency and cross region costs from the connects internal topics...

Another consideration is if we want to mirror to 2 or more separate targets. In that case would it be ideal to create a separate deployment of MM2 for each target Kafka?

If anyone knows about a doc / blog or something with info on this matter or if anyone can give his own 2 cents it would be appreciated!

**Comment by scholzj** [comment, marked as answer]
* Ideally, you should always run the Mirror Maker as close to the target as possible. It is good because of the underlying Connect, but also to minimize the amount of duplicates in the target cluster. I don't think the cost difference in the data transfers would be huge, but there will be some for sure as well if the underlying Connect traffic is within the region.
* You can have multiple sources in one Mirror Maker 2, but only one target. So to have different targets, you will need different Mirror Maker 2 instances.

**Reply by iftachbyaf** [reply]
Hi again Jakub. Thanks for the answers. That seems to align with my conclusions as well the little info I managed to find online on this
